### PR TITLE
Annotate test failures with extra debugging data

### PR DESCRIPTION
**Pull request by nbrahms**
Should help speed up RCA

### CODE REVIEW

File: scripts/test-semgrep-app-e2e.py [hunk 1]
```diff
@@ -44,7 +44,7 @@ def gh_create_branch(branch: str) -> None:
         resp.status_code == HTTPStatus.UNPROCESSABLE_ENTITY
         and resp.json()["message"] == "Reference already exists"
     ):
-        raise Exception(f"Repo {repo_full_name} already has a branch named {branch}")
+        raise Exception(f"Repo {REPO} already has a branch named {branch}")
```
Comment (nbrahms) [inline scripts/test-semgrep-app-e2e.py:47]: `repo_full_name` wasn't declared anywhere

File: scripts/test-semgrep-app-e2e.py [hunk 2]
```diff
@@ -224,7 +230,7 @@ def run_tests():
     # Create a new PR with a known eqeq bug
     new_branch = create_branch(run_id)
     update_file(run_id)
-    pr_id = open_pr(run_id)
+    pr_id = int(open_pr(run_id))
```
Comment (nbrahms) [inline scripts/test-semgrep-app-e2e.py:233]: Resolves a type error

File: scripts/test-semgrep-app-e2e.py (hunk 1 above)
Comment (chmccreery) [reply scripts/test-semgrep-app-e2e.py:47]: good catch! copy/paste error from semgrep_app github.py I think

File: scripts/test-semgrep-app-e2e.py [hunk 3]
```diff
@@ -157,7 +157,13 @@ def notify_sentry(message: str, level: str):
         environment="prod",
     )
     sentry_sdk.set_level(level)
-    sentry_sdk.capture_message(message)
+
+    pr_url = f"https://github.com/semgrep/test-end-to-end-app/pull/{pr_id}"
+
+    with sentry_sdk.push_scope() as scope:
+        scope.set_extra("pr_id", pr_id)
```
Comment (chmccreery) [inline scripts/test-semgrep-app-e2e.py:164]: looks good! Where can I see the scope info show up in sentry?

**Review by chmccreery — APPROVED** [review]

File: scripts/test-semgrep-app-e2e.py (hunk 3 above)
Comment (nbrahms) [reply scripts/test-semgrep-app-e2e.py:164]: that is a good question :)

**Review by DrewDennison — APPROVED** [review]

**Pull request merged**